examples/webgl_instancing_modified.html

```diff
@@ -0,0 +1,196 @@
+
+			import Stats from './jsm/libs/stats.module.js';
+
+			var camera, scene, renderer, stats;
+
+			var mesh;
+
+			var dummy = new THREE.Object3D();
+
+			var amount = 8;
+			var count = Math.pow( amount, 3 );
+
+			init();
+			animate();
+
+			function init() {
+
+				renderer = new THREE.WebGLRenderer( { antialias: false } ); // false improves the frame rate
+				renderer.setPixelRatio( window.devicePixelRatio );
+				renderer.setSize( window.innerWidth, window.innerHeight );
+				document.body.appendChild( renderer.domElement );
+
+				renderer.outputEncoding = THREE.sRGBEncoding;
+
+				scene = new THREE.Scene();
+
+				camera = new THREE.PerspectiveCamera( 40, window.innerWidth / window.innerHeight, 1, 100 );
+				camera.position.set( 0, 0, 20 );
+
+				new THREE.BufferGeometryLoader().load( 'models/json/suzanne_buffergeometry.json', function ( geometry ) {
+
+					var instanceColors = [];
+
+					for ( var i = 0; i < count; i ++ ) {
+
+						instanceColors.push( 0.5 + 0.5 * Math.random() );
+						instanceColors.push( 0.5 + 0.5 * Math.random() );
+						instanceColors.push( 0.5 + 0.5 * Math.random() );
+
+					}
+
+					geometry.setAttribute( 'instanceColor', new THREE.InstancedBufferAttribute( new Float32Array( instanceColors ), 3 ) );
+
+					geometry.computeVertexNormals();
+
+					geometry.scale( 0.5, 0.5, 0.5 );
+
+					//console.log( geometry );
+
+					//
+
+					new THREE.TextureLoader().load( 'textures/matcaps/matcap-porcelain-white.jpg', function ( texture ) {
+
+						texture.encoding = THREE.sRGBEncoding;
+
+						var material = new THREE.MeshMatcapMaterial( { color: 0xaaaaff, matcap: texture } );
+
+						var colorParsChunk = [
+							'attribute vec3 instanceColor;',
+							'varying vec3 vInstanceColor;',
+							'#include <common>'
+						].join( '\n' );
+
+						var instanceColorChunk = [
+							'#include <begin_vertex>',
+							'\tvInstanceColor = instanceColor;'
+						].join( '\n' );
+
+						var fragmentParsChunk = [
+							'varying vec3 vInstanceColor;',
+							'#include <common>'
+						].join( '\n' );
+
+						var colorChunk = [
+							'vec4 diffuseColor = vec4( diffuse * vInstanceColor, opacity );'
+						].join( '\n' );
+
+						material.onBeforeCompile = function ( shader ) {
+
+							shader.vertexShader = shader.vertexShader
+								.replace( '#include <common>', colorParsChunk )
+								.replace( '#include <begin_vertex>', instanceColorChunk );
+
+							shader.fragmentShader = shader.fragmentShader
+								.replace( '#include <common>', fragmentParsChunk )
+								.replace( 'vec4 diffuseColor = vec4( diffuse, opacity );', colorChunk );
+
+							//console.log( shader.uniforms );
+							//console.log( shader.vertexShader );
+							//console.log( shader.fragmentShader );
+
+						};
+
+						mesh = new THREE.InstancedMesh( geometry, material, count );
+
+						mesh.instanceMatrix.setUsage( THREE.DynamicDrawUsage ); // will be updated every frame
```

Comment:
Oh! We've omitted this in `instancing / scatter` and `instancing / dynamic`. The documentation describes `.instanceMatrix` as "For internal use only," which might discourage users from doing this themselves. Should we (a) document this `.setUsage` suggestion, or (b) just enable it by default with InstancedMesh?

Reply:
I would not have described it as "for internal use only".

I'd document the set usage suggestion.

Reply:
I vote for option (b).

Reply:
Remove "internal use only" and document option (a).

I think what I have done in this example is highly appropriate.  I even included a comment for clarification.

Users should set `DynamicDrawUsage` for any attribute that is updated every frame. The default should remain `StaticDrawUsage`.

Instancing is appropriate for static objects such as trees. I would hope that spinning monkeys is not a typical use case.

Reply:
Sounds good to me.